pkg/store/copr/coprocessor.go

```diff
@@ -915,6 +918,10 @@ func (sender *copIteratorTaskSender) run(connID uint64) {
 	if sender.respChan != nil {
 		close(sender.respChan)
 	}
+	if checker != nil {
```

Comment:
No need to check since `ResetTotalProcessedKeys` will do it also.

Reply:
struct's checker field is of type `RunawayChecker`, which is an interface, but it hasn't been initialized with a instance. When calling `checker.ResetTotalProcessedKeys(),` the value of `checker` is nil, leading to a runtime error: invalid memory address or nil pointer dereference.

You can check this [playground](https://go.dev/play/p/MF9M51vdLbo)

Instead of the following approach, maybe we can judge in `ResetTotalProcessedKeys`?

```bash
if checker != nil {
    if rcChecker, ok := checker.(*Checker); ok && rcChecker != nil {
        rcChecker.ResetTotalProcessedKeys()
	}
```